command/agent.go

```diff
@@ -420,7 +425,17 @@ func (c *AgentCommand) Run(args []string) int {
 	// confuse the issue of retries for auth failures which have their own
 	// config and are handled a bit differently.
 	if os.Getenv(api.EnvVaultMaxRetries) == "" {
-		client.SetMaxRetries(config.Vault.Retry.NumRetries)
+		retries := config.Vault.Retry.NumRetries
+
+		// Overrides here to maintain backward compatibility in behavior
+		switch config.Vault.Retry.NumRetries {
+		case -1:
+			retries = 0
+		case 0:
+			retries = defaultClientRetryAttempts
```

Comment:
I'm tempted to change this to be `api.Client` default of 2 retries here (for a total of 3 calls). Even though this is the current value, the client actually times out before making all 12 attempts due to the default request timeout value of 60s.

```
[WARN] (view) vault.read(database/creds/readonly): vault.read(database/creds/readonly): context deadline exceeded (retry attempt 1 after "250ms")
[TRACE] (view) vault.read(database/creds/readonly) starting fetch
[TRACE] vault.read(database/creds/readonly): starting renewer
2021-05-26T15:54:39.425-0700 [INFO]  cache: received request: method=PUT path=/v1/sys/leases/renew
2021-05-26T15:54:39.425-0700 [DEBUG] cache.leasecache: forwarding request: method=PUT path=/v1/sys/leases/renew
2021-05-26T15:54:39.425-0700 [INFO]  cache.apiproxy: forwarding request: method=PUT path=/v1/sys/leases/renew
[WARN] vault.read(database/creds/readonly): failed to renew: context deadline exceeded
[WARN] vault.read(database/creds/readonly): renewer done (maybe the lease expired)
[TRACE] vault.read(database/creds/readonly): GET /v1/database/creds/readonly
2021-05-26T15:55:39.428-0700 [INFO]  cache: received request: method=GET path=/v1/database/creds/readonly
2021-05-26T15:55:39.428-0700 [DEBUG] cache.leasecache: forwarding request: method=GET path=/v1/database/creds/readonly
2021-05-26T15:55:39.428-0700 [INFO]  cache.apiproxy: forwarding request: method=GET path=/v1/database/creds/readonly
[WARN] (view) vault.read(database/creds/readonly): vault.read(database/creds/readonly): context deadline exceeded (retry attempt 2 after "500ms")
[TRACE] (view) vault.read(database/creds/readonly) starting fetch
[TRACE] vault.read(database/creds/readonly): starting renewer
```

Another suggestion that was made was to break retry values by the cache's client and the template to be different values, e.g. `vault.retry.cache_retries` and `vault.retry.template_retries`. We'd still have to support the existing `num_retries` even if we deprecate it though.

Reply:
I'm OK with this change

Reply:
I've gone ahead an made this change on 03d248f, and also updated the table on the PR description.